### PR TITLE
Move defender_atp to new GA package microsoft_defender_endpoint 

### PR DESCRIPTION
**Pull request by adriansr**
## What does this PR do?

- Moves the microsoft.defender_atp datastream into a new package: `microsoft_defender_endpoint`. This new package is GA.
- Reverts #1765, as this instead renamed and GA'd the defender_atp datastream, which we plan to deprecate.
- Copies the new system tests introduced by #1765

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/master/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] If I'm introducing a new feature, I have modified the Kibana version constraint in my package's `manifest.yml` file to point to the latest Elastic stack release (e.g. `^7.13.0`).

## Related issues

-

Relates #1562

## Screenshots

Screenshot 2021-09-29 at 21-11-48 Microsoft Defender for Endpoint - Integrations - Elastic

Screenshot 2021-09-29 at 21-09-16 Add integration - Microsoft Defender for Endpoint - Integrations - Elastic

### CODE REVIEW

File: packages/microsoft_defender_endpoint/_dev/build/build.yml [hunk 1]
```diff
@@ -0,0 +1,3 @@
+dependencies:
+  ecs:
+    reference: git@1.11
```
Comment (P1llus) [inline packages/microsoft_defender_endpoint/_dev/build/build.yml:3]: Could we bump this to 1.12?

File: packages/microsoft_defender_endpoint/changelog.yml [hunk 2]
```diff
@@ -0,0 +1,6 @@
+# newer versions go on top
+- version: "1.0.0"
+  changes:
+    - description: First version
+      type: enhancement
+      link: https://github.com/elastic/integrations/pull/NNN
```
Comment (P1llus) [inline packages/microsoft_defender_endpoint/changelog.yml:6]: We need a link to the github issue here

File: packages/microsoft_defender_endpoint/data_stream/log/elasticsearch/ingest_pipeline/default.yml [hunk 3]
```diff
@@ -0,0 +1,346 @@
+---
+description: Pipeline for parsing Microsoft Defender for Endpoint logs
+processors:
+  - set:
+      field: event.ingested
+      value: '{{_ingest.timestamp}}'
+  - set:
+      field: ecs.version
+      value: '1.11.0'
```
Comment (P1llus) [inline packages/microsoft_defender_endpoint/data_stream/log/elasticsearch/ingest_pipeline/default.yml:9]: Could we bump this to 1.12?

File: packages/microsoft_defender_endpoint/manifest.yml [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+description: This Elastic integration collects logs from Microsoft Defender for Endpoint
+categories:
+  - "network"
+  - "security"
+  - "azure"
+release: ga
+license: basic
+type: integration
+conditions:
+  kibana.version: "^7.14.1"
```
Comment (P1llus) [inline packages/microsoft_defender_endpoint/manifest.yml:14]: I am a bit unsure if we want to bump this to 7.16.0, same with the deprecation of the old one?

**Review by P1llus — APPROVED** [review]
Some small comments, feel free to ignore any that you think is not necessary :)

File: packages/microsoft_defender_endpoint/_dev/build/build.yml (hunk 1 above)
Comment (adriansr) [reply packages/microsoft_defender_endpoint/_dev/build/build.yml]: done

File: packages/microsoft_defender_endpoint/changelog.yml (hunk 2 above)
Comment (adriansr) [reply packages/microsoft_defender_endpoint/changelog.yml]: done

File: packages/microsoft_defender_endpoint/data_stream/log/elasticsearch/ingest_pipeline/default.yml (hunk 3 above)
Comment (adriansr) [reply packages/microsoft_defender_endpoint/data_stream/log/elasticsearch/ingest_pipeline/default.yml]: done

File: packages/microsoft_defender_endpoint/manifest.yml (hunk 4 above)
Comment (adriansr) [reply packages/microsoft_defender_endpoint/manifest.yml:14]: for now we will keep this set to the latest compatible version of kibana

**Review by leehinman — APPROVED** [review]
LGTM.  Thanks for fixing.

**Pull request merged**